### PR TITLE
Refactor findExisting & findAvailable relation routes

### PR DESCRIPTION
**Pull request by petersg83**
### What does it do?

Change the route and controller logic for findExisting and findAvailable relation routes

### CODE REVIEW

File: packages/core/content-manager/server/routes/admin.js [hunk 1]
```diff
@@ -84,19 +84,15 @@ module.exports = {
...
-        ],
+        policies: ['admin::isAuthenticatedAdmin'],
+      },
+    },
+    {
+      method: 'GET',
+      path: '/relations/:model/:id/:targetField',
+      handler: 'relations.findExisting',
+      config: {
+        policies: ['admin::isAuthenticatedAdmin'],
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/routes/admin.js:95]: How will we syncronize with frontend to merge this? As it breaks the content manager with his change.

We will merge https://github.com/strapi/strapi/pull/14544 with this one at the same time?

File: packages/core/content-manager/server/controllers/relations.js [hunk 2]
```diff
@@ -26,57 +26,52 @@ module.exports = {
...
 
-      if (!component && permissionChecker.cannot.read(entity, targetField)) {
+      if (permissionChecker.cannot.read(null, targetField)) {
         return ctx.forbidden();
       }
-      // TODO: find a way to check field permission for component
-      if (component && permissionChecker.cannot.read(entity)) {
-        return ctx.forbidden();
+
+      if (entityId) {
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/controllers/relations.js:55]: shouldn't the entityID be mandatory if it is not a component? The user could make the same request without the entity ID and

Comment (Marc-Roig) [reply packages/core/content-manager/server/controllers/relations.js:55]: Also, I can not edit role permissions,I get a Validation error 
"Permissions have to be defined all together for a subject field or not at all"

Am I missing some step?

Comment (Marc-Roig) [reply packages/core/content-manager/server/controllers/relations.js:55]: If we make it mandatory, we could merge this rbac logic from both endpoints, something like `checkRBACPermissions`

File: packages/core/content-manager/server/controllers/relations.js [hunk 3]
```diff
@@ -132,54 +127,48 @@ module.exports = {
...
+        userAbility,
+        model,
+      });
 
-    if (!entity) {
-      return ctx.notFound();
-    }
+      if (permissionChecker.cannot.read(null, targetField)) {
+        return ctx.forbidden();
+      }
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/controllers/relations.js:153]: Do you know if there is any documentation of the permissionChecker we could look?

File: packages/core/content-manager/server/routes/admin.js (hunk 1 above)
Comment (petersg83) [reply packages/core/content-manager/server/routes/admin.js:95]: yes :)

File: packages/core/content-manager/server/controllers/relations.js (hunk 2 above)
Comment (petersg83) [reply packages/core/content-manager/server/controllers/relations.js:55]: When you create an entry, you need to use this route and you have no entityId to provide. That's why it's optional :)

Regarding the permissions error you have, I think you have an issue with your license/environment. We can check that !

File: packages/core/content-manager/server/controllers/relations.js (hunk 3 above)
Comment (petersg83) [reply packages/core/content-manager/server/controllers/relations.js:153]: Not that I'm aware of. You can ask JS

**Pull request merged**